cou/utils/app_utils.py

```diff
@@ -44,9 +44,59 @@ async def upgrade_packages(units: Iterable[str], model: COUModel) -> None:
             if str(result["Code"]) == "0":
                 logger.debug(result["Stdout"])
             else:
-                raise PackageUpgradeError(
-                    f"Cannot upgrade packages on {unit}."
-                ) from CommandRunFailed(cmd=command, result=result)
+                raise RunUpgradeError(f"Cannot upgrade packages on {unit}.") from CommandRunFailed(
+                    cmd=command, result=result
+                )
 
         except JujuError as exc:
-            raise PackageUpgradeError(f"Cannot upgrade packages on {unit}.") from exc
+            raise RunUpgradeError(f"Cannot upgrade packages on {unit}.") from exc
+
+
+async def set_require_osd_release_option(unit: str, model: COUModel, ceph_release: str) -> None:
+    """Check and set correct value for require-osd-release on a ceph-mon unit.
+
+    :param unit: The ceph-mon unit name where the check command runs on.
+    :type str
+    :param model: COUModel object
+    :type model: COUModel
+    :param ceph_release: The ceph release to set for require-osd-release.
+    :type str
+    :raises RunUpgradeError: When an upgrade fails.
+    """
+    check_command = "ceph osd dump | grep require_osd_release"
```

Comment:
(minor) in order to limit potential issues with invoking chains of shell commands, you could only run `ceph osd dump` and parse the output in python

Reply:
Updated.